doggos/inference/inference_system.py

```diff
@@ -1,19 +1,19 @@
 from abc import ABC, abstractmethod
-from collections import Iterable as Iter
-from typing import Callable, Iterable
+from collections.abc import Iterable as Iter
```

Comment:
IMO a really bad idea to import type as sth. I Would prefer using "from collections.abc import Iterable". You also do not need to use import from typing.